src/features/locators/components/LocatorEditDialog.tsx

```diff
@@ -106,44 +186,50 @@ export const LocatorEditDialog: React.FC<Props> = ({
             isCustomName: isEditedName,
           })
         );
-
+      })
+      .catch((error) => console.log(error))
+      .finally(() => {
         form.resetFields();
         setIsModalOpen(false);
-      })
-      .catch((error) => console.log(error));
+      });
   };
 
-  const renderValidationWarning = () =>
-    !isValidationEnabled ? (
-      <div className="jdn__locatorEdit-warning">
-        <Icon component={WarningFilled} className="ant-alert-icon" />
-        <Footnote>Validation is possible only on Page Object creation</Footnote>
-      </div>
-    ) : null;
+  const renderValidationWarning = () => {
+    if (isCreatingForm && isValidationEnabled) {
+      return (
+        <div className="jdn__locatorEdit-warning">
+          <Icon component={WarningFilled} className="ant-alert-icon" />
+          <Footnote>If you leave this field empty, the locator will be invalid</Footnote>
+        </div>
+      );
+    } else if (!isValidationEnabled) {
+      return (
+        <div className="jdn__locatorEdit-warning">
+          <Icon component={WarningFilled} className="ant-alert-icon" />
+          <Footnote>Validation is possible only on Page Object creation</Footnote>
+        </div>
+      );
+    } else return null;
+  };
 
   return (
     <DialogWithForm
       modalProps={{
-        title: "Edit locator",
+        title: isCreatingForm ? "Create custom locator" : "Edit locator",
         open: isModalOpen,
-        onOk: handleOk,
+        onOk: isCreatingForm ? handleCreateCustomLocator : handleOk,
         enableOverlay: isModalOpen,
         setIsModalOpen,
       }}
       formProps={{
         form,
-        initialValues: {
-          type,
-          name,
-          locator: getXPathByPriority(locator),
-          locatorType: locatorType || pageObjectLocatorType || LocatorType.xPath,
-        },
+        initialValues,
       }}
     >
-      <Form.Item name="name" label="Name:" rules={nameValidationRules}>
+      <Form.Item name="name" label="Name" rules={nameValidationRules}>
         <Input onChange={handleNameChange} />
       </Form.Item>
-      <Form.Item name="type" label="Block type:">
+      <Form.Item name="type" label="Block type" required={isCreatingForm}>
```

Comment:
I think we can make this filed required for all cases

Reply:
done